aries_vcx/src/protocols/common.rs

```diff
@@ -0,0 +1,31 @@
+use crate::messages::error::ProblemReport;
+
+pub fn build_problem_report_msg(comment: Option<String>, thread_id: &str) -> ProblemReport {
```

Comment:
A benefit of using `Option<String>` is that it gives the user flexibility to avoid cloning unnecessarily in cases when they already have a `String` or `Option<String>` to work with; but it seems rather rare in our case... Otherwise, when cloning is necessary and the user usually has a reference, I think it is better done at "the lower level"... I wonder what are your thoughts here?

Reply:
my thoughts in the reply above